From ef08278dd37080a1c04753c5bd6b0b8b80ea06fc Mon Sep 17 00:00:00 2001
From: huangqiwu <huangqiwu@xiaomi.com>
Date: Mon, 11 Nov 2013 10:35:21 +0800
Subject: [PATCH] Fix cdma SignalStrength display

N/A

Signed-off-by: huangqiwu <huangqiwu@xiaomi.com>
---
 .../java/android/telephony/SignalStrength.java     | 31 +++++++++++-----------
 1 file changed, 15 insertions(+), 16 deletions(-)

diff --git a/telephony/java/android/telephony/SignalStrength.java b/telephony/java/android/telephony/SignalStrength.java
index 7b9cf4e..1c809f0 100644
--- a/telephony/java/android/telephony/SignalStrength.java
+++ b/telephony/java/android/telephony/SignalStrength.java
@@ -485,8 +485,8 @@ public int getLevel() {
                 /* We don't know cdma, use evdo */
                 level = evdoLevel;
             } else {
-                /* We know both, use the lowest level */
-                level = cdmaLevel < evdoLevel ? cdmaLevel : evdoLevel;
+                /* We know both, use the highest level */
+                level = cdmaLevel > evdoLevel ? cdmaLevel : evdoLevel;
             }
         }
         if (DBG) log("getLevel=" + level);
@@ -516,8 +516,8 @@ public int getAsuLevel() {
                 /* We don't know cdma use, evdo */
                 asuLevel = evdoAsuLevel;
             } else {
-                /* We know both, use the lowest level */
-                asuLevel = cdmaAsuLevel < evdoAsuLevel ? cdmaAsuLevel : evdoAsuLevel;
+                /* We know both, use the highest level */
+                asuLevel = cdmaAsuLevel > evdoAsuLevel ? cdmaAsuLevel : evdoAsuLevel;
             }
         }
         if (DBG) log("getAsuLevel=" + asuLevel);
@@ -542,8 +542,7 @@ public int getDbm() {
             int cdmaDbm = getCdmaDbm();
             int evdoDbm = getEvdoDbm();
 
-            return (evdoDbm == -120) ? cdmaDbm : ((cdmaDbm == -120) ? evdoDbm
-                    : (cdmaDbm < evdoDbm ? cdmaDbm : evdoDbm));
+            return cdmaDbm;
         }
         if (DBG) log("getDbm=" + dBm);
         return dBm;
@@ -616,10 +615,10 @@ public int getCdmaLevel() {
         int levelDbm;
         int levelEcio;
 
-        if (cdmaDbm >= -75) levelDbm = SIGNAL_STRENGTH_GREAT;
-        else if (cdmaDbm >= -85) levelDbm = SIGNAL_STRENGTH_GOOD;
-        else if (cdmaDbm >= -95) levelDbm = SIGNAL_STRENGTH_MODERATE;
-        else if (cdmaDbm >= -100) levelDbm = SIGNAL_STRENGTH_POOR;
+        if (cdmaDbm >= -85) levelDbm = SIGNAL_STRENGTH_GREAT;
+        else if (cdmaDbm >= -95) levelDbm = SIGNAL_STRENGTH_GOOD;
+        else if (cdmaDbm >= -105) levelDbm = SIGNAL_STRENGTH_MODERATE;
+        else if (cdmaDbm >= -110) levelDbm = SIGNAL_STRENGTH_POOR;
         else levelDbm = SIGNAL_STRENGTH_NONE_OR_UNKNOWN;
 
         // Ec/Io are in dB*10
@@ -629,7 +628,7 @@ public int getCdmaLevel() {
         else if (cdmaEcio >= -150) levelEcio = SIGNAL_STRENGTH_POOR;
         else levelEcio = SIGNAL_STRENGTH_NONE_OR_UNKNOWN;
 
-        int level = (levelDbm < levelEcio) ? levelDbm : levelEcio;
+        int level = levelDbm;
         if (DBG) log("getCdmaLevel=" + level);
         return level;
     }
@@ -676,10 +675,10 @@ public int getEvdoLevel() {
         int levelEvdoDbm;
         int levelEvdoSnr;
 
-        if (evdoDbm >= -65) levelEvdoDbm = SIGNAL_STRENGTH_GREAT;
-        else if (evdoDbm >= -75) levelEvdoDbm = SIGNAL_STRENGTH_GOOD;
-        else if (evdoDbm >= -90) levelEvdoDbm = SIGNAL_STRENGTH_MODERATE;
-        else if (evdoDbm >= -105) levelEvdoDbm = SIGNAL_STRENGTH_POOR;
+        if (evdoDbm >= -85) levelEvdoDbm = SIGNAL_STRENGTH_GREAT;
+        else if (evdoDbm >= -95) levelEvdoDbm = SIGNAL_STRENGTH_GOOD;
+        else if (evdoDbm >= -105) levelEvdoDbm = SIGNAL_STRENGTH_MODERATE;
+        else if (evdoDbm >= -110) levelEvdoDbm = SIGNAL_STRENGTH_POOR;
         else levelEvdoDbm = SIGNAL_STRENGTH_NONE_OR_UNKNOWN;
 
         if (evdoSnr >= 7) levelEvdoSnr = SIGNAL_STRENGTH_GREAT;
@@ -688,7 +687,7 @@ public int getEvdoLevel() {
         else if (evdoSnr >= 1) levelEvdoSnr = SIGNAL_STRENGTH_POOR;
         else levelEvdoSnr = SIGNAL_STRENGTH_NONE_OR_UNKNOWN;
 
-        int level = (levelEvdoDbm < levelEvdoSnr) ? levelEvdoDbm : levelEvdoSnr;
+        int level = levelEvdoDbm;
         if (DBG) log("getEvdoLevel=" + level);
         return level;
     }
-- 
1.9.1

