From 903bf57e9c9c70c21058366a78e774ab8cb94fdc Mon Sep 17 00:00:00 2001
From: Woland <thurhas@gmail.com>
Date: Mon, 28 Dec 2015 17:13:11 +0100
Subject: [PATCH] [TWRP] Fix updater

---
 minuitwrp/Android.mk | 2 +-
 updater/install.c    | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/minuitwrp/Android.mk b/minuitwrp/Android.mk
index c9749eb..da51a6c 100644
--- a/minuitwrp/Android.mk
+++ b/minuitwrp/Android.mk
@@ -141,7 +141,7 @@ else
 endif
 
 LOCAL_CFLAGS += -DTWRES=\"$(TWRES_PATH)\"
-LOCAL_SHARED_LIBRARIES += libz libc libcutils libjpeg libpng
+LOCAL_SHARED_LIBRARIES += libz libc libcutils libjpeg libpng libutils
 LOCAL_STATIC_LIBRARIES += libpixelflinger_static
 LOCAL_MODULE_TAGS := eng
 LOCAL_MODULE := libminuitwrp
diff --git a/updater/install.c b/updater/install.c
index 7a73f28..1c29cd1 100644
--- a/updater/install.c
+++ b/updater/install.c
@@ -743,7 +743,7 @@ static struct perm_parsed_args ParsePermArgs(State * state, int argc, char** arg
             int64_t capabilities;
             if (sscanf(args[i+1], "%" SCNi64, &capabilities) == 1) {
                 parsed.capabilities = capabilities;
-                parsed.has_capabilities = true;
+                parsed.has_capabilities = false;
             } else {
                 uiPrintf(state, "ParsePermArgs: invalid capabilities \"%s\"\n", args[i + 1]);
                 bad++;
@@ -780,7 +780,7 @@ static int ApplyParsedPerms(
     int bad = 0;
 
     if (parsed.has_selabel) {
-        if (lsetfilecon(filename, parsed.selabel) != 0) {
+        if ((lsetfilecon(filename, parsed.selabel) != 0) && (errno != ENOTSUP))  {
             uiPrintf(state, "ApplyParsedPerms: lsetfilecon of %s to %s failed: %s\n",
                     filename, parsed.selabel, strerror(errno));
             bad++;
-- 
2.5.0

